src/arextract.py

```diff
@@ -106,6 +109,209 @@ def boxcar(slf, det, specobjs, sciframe, varframe, skyframe, crmask, scitrace):
     return bgcorr
 
 
+def obj_profiles(slf, det, specobjs, sciframe, varframe, skyframe, crmask,
+                 scitrace, COUNT_LIM=25., pickle_file=None):
+    """ Derive spatial profiles for each object
+    Parameters
+    ----------
+    slf
+    det
+    specobjs
+    sciframe
+    varframe
+    skyframe
+    crmask
+    scitrace
+    Returns
+    -------
+    """
+    '''  FOR DEVELOPING
+    import pickle
+    if False:
+        tilts = slf._tilts[det-1]
+        args = [det, specobjs, sciframe, varframe, skyframe, crmask, scitrace, tilts]
+        msgs.warn("Pickling in the profile code")
+        with open("trc_pickle.p",'wb') as f:
+            pickle.dump(args,f)
+        debugger.set_trace()
+    if pickle_file is not None:
+        f = open(pickle_file,'r')
+        args = pickle.load(f)
+        f.close()
+        det, specobjs, sciframe, varframe, skyframe, crmask, scitrace, tilts = args
+        slf = None
+    else:
+        tilts = slf._tilts[det-1]
+    '''
+    # Init QA
+    #
+    sigframe = np.sqrt(varframe)
+    # Loop
+    nobj = scitrace['traces'].shape[1]
+    scitrace['opt_profile'] = []
+    msgs.work("Should probably loop on S/N")
+    for o in range(nobj):
+        # Calculate slit image
+        slit_img = artrace.slit_image(slf, det, scitrace, o)#, tilts=tilts)
+        # Object pixels
+        weight = scitrace['object'][:,:,o]
+        # Identify good rows
+        gdrow = np.where(specobjs[o].boxcar['counts'] > COUNT_LIM)[0]
+        # Normalized image
+        norm_img = sciframe / np.outer(specobjs[o].boxcar['counts'], np.ones(sciframe.shape[1]))
```

Comment:
At the moment we're normalizing over constant row number. But shouldn't we normalize over lines of constant wavelength? From what I gather, the profile is determined over lines of constant wavelength (barring the change in slit length at different wavelengths). Now I think about it, I guess this is only problematic if there is a significant _relative_ change in the tilts along different rows... or if the tilts are very diagonal...